src/Main.java

```diff
@@ -2251,11 +2259,20 @@ else if (region == 0x12) {
 			&& (difficulty > Difficulty.EASY || inventory.contains(Items.JUMP_BOOTS));
             }
             else if (region == 0x15) {
+				// New MINOR GLITCHES Logic for Inside 2nd Hill.
+				// You can skip needing Boots if you have Red Overalls by breaking the blocks in a specific way, and then doing a walljump after obtaining this check.
+				// This can only be in logic on MINOR GLITCHES if it specifically contains the Red Key.
                 return inventory.contains(Items.GARLIC) && canLift(inventory)
-                        && (canSwim(inventory) || (difficulty >= Difficulty.HARD && inventory.contains(Items.JUMP_BOOTS)));
+                        && (canSwim(inventory) 
+							|| ((difficulty >= Difficulty.HARD && inventory.contains(Items.JUMP_BOOTS)) 
+							|| (difficulty >= Difficulty.S_HARD && canSuperGP(inventory)));
             }
             else if (region == 0x16) {
-                return inventory.contains(Items.GARLIC) && (canSwim(inventory) || (difficulty >= Difficulty.S_HARD && inventory.contains(Items.JUMP_BOOTS)));
+				// New MERCILESS Logic for Inside 4th Hill.
+				// You can skip needing Boots if you have Red Overalls by breaking the blocks in a specific way, and then doing a reverse walljump after obtaining this check.
+				// The required 
```

Comment:
Looks like this comment didn't get finished

Reply:
Fixed. Comment is supposed to say "The required single-tile walljump takes place at the 3rd hill, which is why this trick needs to be Merciless without Boots."